### PR TITLE
Rework/charts rework

### PR DESCRIPTION
**Pull request by yesod30**
Rewrite of charts to make use of ship builds. This solves #194 and also enables acceleration charts for all users, ending the beta period.

### CODE REVIEW

**Review by iTTou — APPROVED** [review]

File: WoWsShipBuilder.Core/DataContainers/DispersionPlotHelper.cs [hunk 1]
```diff
@@ -26,15 +26,15 @@ private static (double horizontalRadius, double verticalRadius) GetDispersionEll
         /// <summary>
         /// Calculates the projection ratio and the vertical radius length of the dispersion ellipse projected on water.
         /// </summary>
-        /// <param name="shell">Contains all the shell parameters.</param>
+        /// <param name="shell">The shell to calculate the <see cref="Ballistic"/> of.</param>
         /// <param name="maxRange">Max range a ship can fire at.</param>
         /// <param name="aimingRange">Range the ship is currently aiming at.</param>
         /// <param name="verticalRadius">The length of the dispersion ellipse vertical radius.</param>
         /// <returns>The projection ratio on water and on the perpendicular plane to the water, the length of the vertical radius projected on water and on the perpendicular plane to the water.</returns>
         private static (double waterLineProjection, double perpendicularToWaterProjection, double projectedOnWaterVerticalRadius, double perpendicularToWaterVerticalRadius) GetProjectedEllipse(ArtilleryShell shell, double maxRange, double aimingRange, double verticalRadius)
```
Comment (floribe2000) [inline WoWsShipBuilder.Core/DataContainers/DispersionPlotHelper.cs:34]: Using tuples with that many parameters may justify using a dedicated type for it

File: WoWsShipBuilder.Web/Components/DispersionPlot.razor [hunk 2]
```diff
@@ -10,16 +10,16 @@
 
 <div class="d-flex ma-auto">
     <MudStack Row="IsVertical" Justify="Justify.Center" AlignItems="AlignItems.Center">
-        @foreach (var dispersionEllipse in Ships)
+        @foreach (var (id, dispersionEllipse) in Ships)
         {
             <MudStack Justify="Justify.Center" AlignItems="AlignItems.Center">
                 <MudText Style="font-size: x-large">
-                    @($"{Localizer.GetGameLocalization($"{dispersionEllipse.Name[..dispersionEllipse.Name.IndexOf(" - ", StringComparison.Ordinal)]}_FULL").Localization} - {Localizer.GetGameLocalization(dispersionEllipse.Shell.Name).Localization}")
+                    @dispersionEllipse.Name
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Components/DispersionPlot.razor:17]: Why does the data structure contain localized data? Is this necessary or not?

File: WoWsShipBuilder.Web/Components/DispersionPlot.razor [hunk 3]
```diff
@@ -39,7 +39,7 @@
 @code {
 
     [Parameter]
-    public List<DispersionEllipse> Ships { get; set; } = default!;
+    public Dictionary<Guid, DispersionEllipse>  Ships { get; set; } = default!;
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Components/DispersionPlot.razor:42]: Don't initialize with "default!" when there is no EditorRequired attribute.
Also check if using ImmutableDictionary may be the right choice here. Ideally, parameters are immutable types

File: WoWsShipBuilder.Web/Components/ShipStatsComponent.razor [hunk 4]
```diff
@@ -467,6 +469,8 @@
 }
 
 @code {
+    [Parameter, EditorRequired]
+    public EventCallback SaveBuildStringIntoLocal { get; set; }
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Components/ShipStatsComponent.razor:473]: Why a callback?

File: WoWsShipBuilder.Web/Dialogs/ShipAndShellSelectionDialog.razor [hunk 5]
```diff
@@ -20,25 +26,31 @@
         <MudContainer Class="d-flex" Style="min-height: 600px; max-height: 600px; overflow-y: auto">
             <MudGrid Justify="Justify.Center">
                 <MudItem xs="12" md="8" Class="d-flex" Style="overflow-y: auto">
-                    <ShipSelector SelectedShips="@selectedShips" LargerList="true"/>
+                    <ShipSelector SelectedShips="@selectedShips" LargerList AllowCopies/>
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Dialogs/ShipAndShellSelectionDialog.razor:29]: Consider adjusting the selector to allow filtering ships that are irrelevant for this dialog (in this case: all ships without main or secondary guns)

**Review by floribe2000 — CHANGES_REQUESTED** [review]
Looks good for the most parts, please provide clarification for comments

**Review by floribe2000 — APPROVED** [review]
LGTM

**Pull request merged**